web/ui/react-app/src/Graph.test.tsx

```diff
@@ -101,5 +101,60 @@ describe('Graph', () => {
       expect(legend).toHaveLength(1);
     });
+    it('formats tick values correctly', () => {
+      const graph = new Graph();
+      expect(graph.formatValue(2e24)).toBe('2.00Y');
+      expect(graph.formatValue(2e23)).toBe('200.00Z');
+      expect(graph.formatValue(2e22)).toBe('20.00Z');
+      expect(graph.formatValue(2e21)).toBe('2.00Z');
+      expect(graph.formatValue(2e19)).toBe('20.00E');
+      expect(graph.formatValue(2e18)).toBe('2.00E');
+      expect(graph.formatValue(2e17)).toBe('200.00P');
+      expect(graph.formatValue(2e16)).toBe('20.00P');
+      expect(graph.formatValue(2e15)).toBe('2.00P');
+      expect(graph.formatValue(1e15)).toBe('1.00P');
+      expect(graph.formatValue(2e14)).toBe('200.00T');
+      expect(graph.formatValue(2e13)).toBe('20.00T');
+      expect(graph.formatValue(2e12)).toBe('2.00T');
+      expect(graph.formatValue(2e11)).toBe('200.00G');
+      expect(graph.formatValue(2e10)).toBe('20.00G');
+      expect(graph.formatValue(2e9)).toBe('2.00G');
+      expect(graph.formatValue(2e8)).toBe('200.00M');
+      expect(graph.formatValue(2e7)).toBe('20.00M');
+      expect(graph.formatValue(2e6)).toBe('2.00M');
+      expect(graph.formatValue(2e5)).toBe('200.00k');
+      expect(graph.formatValue(2e4)).toBe('20.00k');
+      expect(graph.formatValue(2e3)).toBe('2.00k');
+      expect(graph.formatValue(2e2)).toBe('200.00');
+      expect(graph.formatValue(2e1)).toBe('20.00');
+      expect(graph.formatValue(2)).toBe('2.00');
+      expect(graph.formatValue(2e-1)).toBe('0.20');
+      expect(graph.formatValue(2e-2)).toBe('0.02');
+      expect(graph.formatValue(2e-3)).toBe('2.00m');
+      expect(graph.formatValue(2e-4)).toBe('0.20m');
+      expect(graph.formatValue(2e-5)).toBe('0.02m');
+      expect(graph.formatValue(2e-6)).toBe('2.00µ');
+      expect(graph.formatValue(2e-7)).toBe('0.20µ');
+      expect(graph.formatValue(2e-8)).toBe('0.02µ');
+      expect(graph.formatValue(2e-9)).toBe('2.00n');
+      expect(graph.formatValue(2e-10)).toBe('0.20n');
+      expect(graph.formatValue(2e-11)).toBe('0.02n');
+      expect(graph.formatValue(2e-12)).toBe('2.00p');
+      expect(graph.formatValue(2e-13)).toBe('0.20p');
+      expect(graph.formatValue(2e-14)).toBe('0.02p');
+      expect(graph.formatValue(2e-15)).toBe('2.00f');
+      expect(graph.formatValue(2e-16)).toBe('0.20f');
+      expect(graph.formatValue(2e-17)).toBe('0.02f');
+      expect(graph.formatValue(2e-18)).toBe('2.00a');
+      expect(graph.formatValue(2e-19)).toBe('0.20a');
+      expect(graph.formatValue(2e-20)).toBe('0.02a');
+      expect(graph.formatValue(1e-21)).toBe('1.00z');
+      expect(graph.formatValue(2e-21)).toBe('2.00z');
+      expect(graph.formatValue(2e-22)).toBe('0.20z');
+      expect(graph.formatValue(2e-23)).toBe('0.02z');
+      expect(graph.formatValue(2e-24)).toBe('2.00y');
+      expect(graph.formatValue(2e-25)).toBe('0.20y');
+      expect(graph.formatValue(2e-26)).toBe('0.02y');
```

Comment:
How about putting the key/value pairs into an object and then mapping over them?

Reply:
Indeed, I have done the change.